drivers/i2c/Kconfig

```diff
@@ -23,6 +23,12 @@ config I2C_SHELL
 	  The I2C shell supports scanning, bus recovery, I2C read and write
 	  operations.
 
+config I2C_ZERO_LEN_XFER_SUPPORTED
+	bool "I2C transfer without data (data length 0)"
+	default y
+	help
+	  Enable this flag to support 0 length I2C transfers(quick commands).
```

Comment:
Quick commands are SMBus, not I2C.

Reply:
As, few of the I2C IPs support Quick Commands based on the hardware configuration (for various i2c-controller and target device combination on various platforms).

Thus, making use of this Kconfig macro we can enable/disable the support of functionalities or commands which are making use of 0 data-length i2c commands.